week-4/E-arrays-of-objects/exercise-1.js

```diff
@@ -5,42 +5,59 @@
 */
 
 var person1 = {
-    name: "Alice",
-    age: 25
+  name: "Alice",
+  age: 25
 };
 
 var person2 = {
-    name: "Bob",
-    age: 30
+  name: "Bob",
+  age: 30
 };
 
 var person3 = {
-    name: "John",
-    age: 20
+  name: "John",
+  age: 20
 };
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var persons = [person1, person2, person3]; // Complete here
 
-var persons = // Complete here
-
-var personNames = // Complete here
-
-var personsYoungerThan28YearsOld = // Complete here
+var personNames = persons.map(person => person.name); // Complete here
 
+var personsYoungerThan28YearsOld = persons.filter(person => {
+  if (person.age < 28) {
+    return person;
+  }
+}); // Complete here
 
 /*
 DO NOT EDIT ANYTHING BELOW THIS LINE
 */
 
-console.log("Question 1: array defined with 3 persons -> ", 
-    (persons[0] === person1 && persons[1] === person2 && persons[2] === person3) ? 'Passed :)' : 'Not yet :(');
-
-console.log("Question 2: array containing the person names -> ", 
-    (personNames[0] === "Alice" && personNames[1] === "Bob" && personNames[2] === "John") ? 'Passed :)' : 'Not yet :(');
-
-console.log("Question 3: array containing the persons younger than 28 years old -> ", 
-    (personsYoungerThan28YearsOld[0] === person1 && personsYoungerThan28YearsOld[1] === person3) ? 'Passed :)' : 'Not yet :(');
+console.log(
+  "Question 1: array defined with 3 persons -> ",
+  persons[0] === person1 && persons[1] === person2 && persons[2] === person3
+    ? "Passed :)"
+    : "Not yet :("
+);
+
+console.log(
+  "Question 2: array containing the person names -> ",
+  personNames[0] === "Alice" &&
+    personNames[1] === "Bob" &&
+    personNames[2] === "John"
+    ? "Passed :)"
+    : "Not yet :("
+);
+
+console.log(
+  "Question 3: array containing the persons younger than 28 years old -> ",
+  personsYoungerThan28YearsOld[0] === person1 &&
+    personsYoungerThan28YearsOld[1] === person3
+    ? "Passed :)"
+    : "Not yet :("
```

Comment:
Good solution - one minor improvement that could be added. Great use of array methods! 👍